app/controllers/chat_channel_memberships_controller.rb

```diff
@@ -115,14 +115,49 @@ def leave_membership
     end
   end
 
+  def update_membership_role
+    @chat_channel = ChatChannel.find_by(id: params[:id])
+    authorize @chat_channel, :update?
+    membership = ChatChannelMembership.find_by(id: channel_membership_params[:membership_id], chat_channel_id: @chat_channel.id)
+
+    membership.update(role: channel_membership_params[:role])
+    if membership.errors.any?
+      render json: { success: false, message: "Failed to update membership", errors: chat_channel_membership.errors.full_messages }, status: :bad_request
+    else
+      render json: { success: true, message: "User Membership is updated" }, status: :ok
+    end
+  end
+
+  def join_channel_invitation
+    chat_channel = ChatChannel.find_by(slug: params[:channel_slug])
+    authorize chat_channel
+    membership = ChatChannelMembership.find_by(user_id: current_user.id, chat_channel_id: chat_channel.id)
+
+    if !membership
+      membership = ChatChannelMembership.new(user_id: current_user.id, chat_channel_id: chat_channel.id)
+      membership.save
+      send_chat_action_message("@#{membership.user.username} join the channel", current_user, chat_channel.id, "joined") unless membership&.errors&.any?
+    elsif membership.status != "active"
+      membership.update(role: "member", status: "active")
+      send_chat_action_message("@#{membership.user.username} join the channel", current_user, membership.chat_channel_id, "joined")
```

Comment:
I think you could refractor this one step further and pull this `send_chat_action_message` out of both sides of this boolean and put it below.

```ruby
if membership&.errors&.any?
  flash[:settings_notice] = membership.errors.full_messages
  redirect_to root_path
else
    send_chat_action_message("@#{membership.user.username} join the channel", current_user, membership.chat_channel_id, "joined")
end

Reply:
@mstruve In that case, If a user is already a member and he used this link. then the joining message will appear for him in chat.

Reply:
Could you solve that by doing this?
```
if membership&.errors&.any?
...ruby
elsif membership.status != "active"
  send_chat_action_message(..)
end
```